yandexcloud/_channels.py

```diff
@@ -30,35 +40,69 @@ def __init__(self, client_user_agent=None, **kwargs):
             endpoint=self._endpoint,
         )
 
-        self._unauthenticated_channel = None
-        self._channels = None
         self._client_user_agent = client_user_agent
-
-    def channel_options(self):
-        return tuple(
+        self._config_endpoints = copy(endpoints)
+        self._endpoints = None
+        self.channel_options = tuple(
             ("grpc.primary_user_agent", user_agent)
             for user_agent in [self._client_user_agent, SDK_USER_AGENT]
             if user_agent is not None
         )
 
-    def channel(self, endpoint):
-        if not self._channels:
-            self._unauthenticated_channel = grpc.secure_channel(
-                self._endpoint, self._channel_creds, options=self.channel_options()
+    def channel(self, service: str, endpoint: Optional[str] = None, insecure: bool = False) -> grpc.Channel:
+        if endpoint:
+            logger.info("Using provided service %s endpoint %s", service, endpoint)
+            if insecure:
+                logger.info("Insecure option is ON, no IAM endpoint used for verification")
+                return grpc.insecure_channel(endpoint, options=self.channel_options)
+            logger.info("Insecure option is OFF,IAM endpoint %s used for verification")
+            creds = self._get_creds(self.endpoints["iam"])
+            return grpc.secure_channel(endpoint, creds, options=self.channel_options)
+        if service not in self._config_endpoints and insecure:
+            logger.warning(
+                "Unable to use insecure option for default {%s} service endpoint.\n"
+                "Option is ignored. To enable it override endpoint.",
+                service,
             )
-            endpoint_service = ApiEndpointServiceStub(self._unauthenticated_channel)
-            resp = endpoint_service.List(ListApiEndpointsRequest())
-            endpoints = resp.endpoints
+        else:
```

Comment:
So if `service in self._config_endpoints` but `insecure=False`, we are falling into this branch.
By log message below it is looks like a bug.

Reply:
fixed